modules/core/src/main/java/org/apache/synapse/MessageContext.java

```diff
@@ -457,4 +457,19 @@ public interface MessageContext {
      * @param tracingState Set whether the message flowtracing is enabled or not
      */
     public void setMessageFlowTracingState(int tracingState);
+
+    /**
+     * Adds a new latency calculation task, where the output can be retrieved with {@link #getLatency()}.
+     * For each {@link #recordLatency()} method call there should be a corresponding {@link #getLatency()} method call.
+     */
+    public default void recordLatency() {};
+
+    /**
+     * Calculates the time between the previous corresponding {@link #recordLatency()} method call and now. This should
+     * be used with {@link #recordLatency()}.
+     * @return latency in milliseconds.
+     */
+    public default long getLatency() {
+        return 0;
+    };
```

Comment:
Since a this is a library and `MessageContext` is a very commonly used low level interface, any changes to this will require a change in all its' dependents which can be nasty. And then again I can move this change to `Axis2MessageContext`, by doing so I have to cast `MessageContext` into `Axis2MessageContext` and then call `recordLatency` and `getLatency`.  This means the below code small code segment will be duplicated all over its usages.
```java
if (synCtx instanceof Axis2MessageContext) {
    ((Axis2MessageContext)synCtx).recordLatency();
}
```
Therefore I decided to go with default implementation in interface level.
@Nirothipan WDYT?

Reply:
+1